drivers/sensor/lps22df/lps22df.c

```diff
@@ -0,0 +1,356 @@
+/* ST Microelectronics LPS22DF pressure and temperature sensor
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lps22df.pdf
+ */
+
+#define DT_DRV_COMPAT st_lps22df
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lps22df.h"
+
+LOG_MODULE_REGISTER(LPS22DF, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int lps22df_set_odr_raw(const struct device *dev, uint8_t odr)
+{
+	const struct lps22df_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
```

Comment:
Do you need the casting?

```c
struct lps22df_config {
	stmdev_ctx_t ctx;
...
```

Reply:
Yes, it's just to avoid this warning:

```
warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   76 |         stmdev_ctx_t *ctx = &cfg->ctx;
```

Reply:
@fabiobaltieri it is something that was highlighted some times ago. In order to fix it I guess I would need to change all the stmemscc module APIs adding const to ctx. Like this:

```
int32_t lps22df_bus_mode_set(const stmdev_ctx_t *ctx, lps22df_bus_mode_t *val)```
```

STdC are something that I can control, but it is a huge change, even if good change (ctx is really not modifiable). I will think about it.

Reply:
got it, yeah not worth it